### DISCUSSION TITLE
How to have a pathbuf where a segment starts with a period

### DISCUSSION THREAD

**Discussion by arthurmelton** [rwf2/Rocket#2355] [Questions] [answered]
So if you use the example at [https://rocket.rs/v0.5-rc/guide/requests/#multiple-segments](https://rocket.rs/v0.5-rc/guide/requests/#multiple-segments) and try and open `127.0.0.1/example` this will work but `127.0.0.1/.example` will not work, is there a way to make it so that segments can have a period at the beginning?

**Comment by edgarogh** [comment, marked as answer]
According to [the docs](https://docs.rs/rocket/0.5.0-rc.2/rocket/request/trait.FromSegments.html#provided-implementations):

> For security purposes, any other segments that begin with “*” or “.” are ignored.

I'm afraid you'll have to make your own newtype around `PathBuf` and implement `FromSegments` on it yourself. It should be really easy though, as you almost just have to copy this...

https://github.com/SergioBenitez/Rocket/blob/6c3d35e7e5bceaf11c329623d48f81c276407030/core/lib/src/request/from_param.rs#L319-L325

...and replace the argument (`allow_dotfiles`) from `false` to `true`. It could look like this:

```rs
struct PathBufWithDotfiles(PathBuf);

impl FromSegments<'_> for PathBufWithDotfiles {
    type Error = PathError;

    fn from_segments(segments: Segments<'_, Path>) -> Result<Self, Self::Error> {
        segments.to_path_buf(true).map(Self)
    }
}
```

_Disclaimer: I didn't test this code._

**Reply by arthurmelton** [reply]
if anyone wants to do this dont forget to import these and not the std versions:

```rust
use std::path::PathBuf;
use rocket::http::uri::{Segments, error::PathError, fmt::Path};
use rocket::request::FromSegments;
```
